sharpy/linear/assembler/lincontrolsurfacedeflector.py

```diff
@@ -166,16 +168,28 @@ def generate(self, linuvlm=None, tsaero0=None, tsstruct0=None, aero=None, struct
                                 with_control_surface = True
```

Comment:
I think you can simplify the code changing lines 117-122 with:

for global_node in range(structure.num_node):
	i_elem, i_local_node = structure.node_master_elem

Reply:
For the moment I need to keep this as is, for the case where a single node is attached to multiple elements. 

However, I do agree that the current implementation is inefficient. It should change to be done on a per-element basis but for the moment this gets the job done ;)